Document/IR.md

```diff
@@ -0,0 +1,105 @@
+# How IR work inside amacc
+
+# Instructsion sets
+
+these instructions is for general purpose
```

Comment:
Before explaining ISA, you should mention the design of stack-based IR. You can read further descriptions via https://github.com/lotabout/write-a-C-interpreter/tree/master/tutorial/en